### ISSUE TITLE
Infinite loop when backwards `f32` range is given.

### ISSUE THREAD

**Issue by siefkenj** [38/plotters#255]
**Describe the bug**
When an `f32` range is given to `build_cartesian_2d` specified from a higher number to a lower number, plotters will get stuck in a loop. This doesn't seem to happen when the range is `i32`.

**To Reproduce**
Modified from the `matshow.rs` example.
```rust
    let mut chart = ChartBuilder::on(&root)
        .caption("Demonstration of full_palette Colors", ("sans-serif", 60))
        .margin(5)
        .top_x_label_area_size(40)
        .y_label_area_size(40)
        .build_cartesian_2d(0f32..15f32, 15f32..0f32)?;
```

The problem is fixed by changing it to
```rust
    let mut chart = ChartBuilder::on(&root)
        .caption("Demonstration of full_palette Colors", ("sans-serif", 60))
        .margin(5)
        .top_x_label_area_size(40)
        .y_label_area_size(40)
        .build_cartesian_2d(0f32..15f32, 0f32..15f32)?;
```

**Version Information**
0.3.1 (git master)

**Comment by 38** [comment]
Thanks for reporting the bug. It should be fixed in master branch after commit 5d4821a.

Please let me know if the fix works.

Thanks!

**Comment by siefkenj** [comment]
Yep. The fix works!
